### PR TITLE
[Relax][Frontend][TFLite] Add DILATE operator mapping

### PR DESCRIPTION
**Pull request by as4230**
This PR adds TFLite frontend support for the DILATE operator which extends a tensor by inserting a padding value between existing elements per axis according to the dilation strides.

Decomposes into existing Relax primitives instead of registering a new op:
applied per axis:
- relax.op.reshape adds a size-1 stride-axis and merges it back after padding
- relax.op.full builds a padding tensor with (stride - 1) values along that axis
- relax.op.concat interleaves the padding between input elements
- relax.op.strided_slice trims the trailing pad to output size

Both static and dynamic dilations are supported.

Frontend tests use hand-rolled .tflite fixtures since DILATE has no public TF Python emitter through tf.lite.TFLiteConverter, so the standard verify(TestClass, Expected) pattern can't reach it. Extends DENSIFY's fixture builders to handle BuiltinOptions2 and non-FLOAT32 tensors. _finish_tflite_model now writes the TFL3 file identifier so the produced buffer is a valid input for tf.lite.Interpreter in the nightly E2E path.

Validation:
    python -m pytest tests/python/relax/test_frontend_tflite.py -k dilate -v

Addresses the DILATE item under #19412.

### CODE REVIEW

File: python/tvm/relax/frontend/tflite/tflite_frontend.py [hunk 1]
```diff
@@ -3416,6 +3417,53 @@ def convert_dequantize(self, op):
...
+        input_tensors = self.get_input_tensors(op)
+        output_tensors = self.get_output_tensors(op)
+        assert len(input_tensors) == 3, "input tensors length should be 3"
+        assert len(output_tensors) == 1, "output tensors length should be 1"
+
+        in_expr = self.get_tensor_expr(input_tensors[0])
+        in_shape = to_int_list(self.get_tensor_shape(input_tensors[0]))
+        in_dtype = self.get_tensor_type_str(input_tensors[0].tensor.Type())
+        n_dims = len(in_shape)
+        n_elements = int(np.prod(in_shape))
```
[the remark this answers is not in the document]
Comment (as4230) [reply python/tvm/relax/frontend/tflite/tflite_frontend.py]: _input_type at tflite_frontend.py:4408 calls tensor.ShapeAsNumpy() and never ShapeSignatureAsNumpy(), so any -1 in shape_signature gets dropped for every op in this frontend and not just DILATE. A normal tf.keras.Input(batch_size=None) Keras model serializes to TFLite as shape=[1, ...], shape_signature=[-1, ...], and TVM imports it hardcoded to batch=1.

Fixing it would touch _input_type, get_tensor_shape, and any handler reading int(shape[i]) so not DILATE scope. Can file a separate issue

**Review by tlopex — COMMENTED** [review]
I don't understand why you did't use `tvm.ir.assert_structural_equal(mod, Expected)` here but e2e

**Comment by as4230** [comment]
> I don't understand why you did't use `tvm.ir.assert_structural_equal(mod, Expected)` here but e2e

@tlopex  I originally went with e2e because it was less verbose and tested runtime behavior but I switched it to tvm.ir.assert_structural_equal against an expected IR per your suggestion.

**Review by tlopex — APPROVED** [review]
